### PR TITLE
vaultwarden: 1.26.0 -> 1.27.0

### PR DESCRIPTION
**Pull request by wahjava**
###### Description of changes

https://github.com/dani-garcia/vaultwarden/releases/tag/1.27.0

To be committed with #207594 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by ivan — APPROVED** [review]
I built and tested this PR in combination with #207594 on x86_64 NixOS 22.11 and I did not see issues.

**Comment by busti** [comment]
Is there a way to expedite this? This is currently preventing vaultwarden from being used with the firefox plugin.

**Comment by dotlambda** [comment]
> To be committed with #207594

Why separate PRs then?

**Comment by wahjava** [comment]
> > To be committed with #207594
> 
> Why separate PRs then?

I'm not that familiar with nixpkgs contribution, and just assumed one derivation per PR.

**Comment by dotlambda** [comment]
> > > To be committed with #207594
> > 
> > 
> > Why separate PRs then?
> 
> I'm not that familiar with nixpkgs contribution, and just assumed one derivation per PR.

You should add commits to a single PR if they make sense together.

File: pkgs/tools/security/vaultwarden/default.nix [hunk 1]
```diff
@@ -9,16 +9,16 @@ in
...
   src = fetchFromGitHub {
     owner = "dani-garcia";
     repo = pname;
     rev = version;
-    sha256 = "sha256-LPIc1odUBvjVJty3GYYFNhile4XBWMisLUeVtWH6xgE=";
+    sha256 = "sha256-QvU1Y3syr6PZbTRebbZF4sEzI4lIj1enJe2F/gGfvQM=";
   };
 
-  cargoSha256 = "sha256-IfseODaoqlPNBlVjS+9+rKXAOq29TgULMA/ogmqg0NA=";
+  cargoSha256 = "sha256-lylRGg5pzJ4sBS3bY4ObMoJ5s5kakMLTtq1VOnmS5HM=";
```
Comment (dotlambda) [inline pkgs/tools/security/vaultwarden/default.nix:21]:
```suggestion
  cargoHash = "sha256-lylRGg5pzJ4sBS3bY4ObMoJ5s5kakMLTtq1VOnmS5HM=";
```

File: pkgs/tools/security/vaultwarden/default.nix [hunk 2]
```diff
@@ -9,16 +9,16 @@ in
...
   pname = "vaultwarden";
-  version = "1.26.0";
+  version = "1.27.0";
 
   src = fetchFromGitHub {
     owner = "dani-garcia";
     repo = pname;
     rev = version;
-    sha256 = "sha256-LPIc1odUBvjVJty3GYYFNhile4XBWMisLUeVtWH6xgE=";
+    sha256 = "sha256-QvU1Y3syr6PZbTRebbZF4sEzI4lIj1enJe2F/gGfvQM=";
```
Comment (dotlambda) [inline pkgs/tools/security/vaultwarden/default.nix:18]:
```suggestion
    hash = "sha256-QvU1Y3syr6PZbTRebbZF4sEzI4lIj1enJe2F/gGfvQM=";
```

File: pkgs/tools/security/vaultwarden/default.nix [hunk 3]
```diff
@@ -9,16 +9,16 @@ in
...
   };
 
-  cargoSha256 = "sha256-IfseODaoqlPNBlVjS+9+rKXAOq29TgULMA/ogmqg0NA=";
+  cargoSha256 = "sha256-lylRGg5pzJ4sBS3bY4ObMoJ5s5kakMLTtq1VOnmS5HM=";
 
   postPatch = ''
     # Upstream specifies 1.57; nixpkgs has 1.56 which also produces a working
```
Comment (dotlambda) [inline pkgs/tools/security/vaultwarden/default.nix:24]: That's not true anymore.

Comment (wahjava) [reply pkgs/tools/security/vaultwarden/default.nix:24]: I still get the same sha256 hash, so I'm not sure what do you mean is not true anymore. Could you please elaborate ?

**Comment by dotlambda** [comment]
Can you fix the NixOS test?

**Comment by rhysmdnz** [comment]
The following patch fixes the tests

```patch
diff --git a/nixos/tests/vaultwarden.nix b/nixos/tests/vaultwarden.nix
index c0e1d0585b9..d0b11e00538 100644
--- a/nixos/tests/vaultwarden.nix
+++ b/nixos/tests/vaultwarden.nix
@@ -107,7 +107,7 @@ let
 
                   wait = WebDriverWait(driver, 10)
 
-                  wait.until(EC.title_contains("Create Account"))
+                  wait.until(EC.title_contains("Create account"))
 
                   driver.find_element(By.CSS_SELECTOR, 'input#register-form_input_email').send_keys(
                       '${userEmail}'
@@ -122,18 +122,20 @@ let
                       '${userPassword}'
                   )
 
-                  driver.find_element(By.XPATH, "//button[contains(., 'Create Account')]").click()
+                  driver.find_element(By.XPATH, "//button[contains(., 'Create account')]").click()
 
-                  wait.until_not(EC.title_contains("Create Account"))
+                  wait.until_not(EC.title_contains("Create account"))
+
+                  driver.find_element(By.XPATH, "//button[contains(., 'Continue')]").click()
 
                   driver.find_element(By.CSS_SELECTOR, 'input#login_input_master-password').send_keys(
                       '${userPassword}'
                   )
-                  driver.find_element(By.XPATH, "//button[contains(., 'Log In')]").click()
+                  driver.find_element(By.XPATH, "//button[contains(., 'Log in')]").click()
 
-                  wait.until(EC.title_contains("Bitwarden Web Vault"))
+                  wait.until(EC.title_contains("Vaultwarden Web Vault"))
 
-                  driver.find_element(By.XPATH, "//button[contains(., 'Add Item')]").click()
+                  driver.find_element(By.XPATH, "//button[contains(., 'Add item')]").click()
 
                   driver.find_element(By.CSS_SELECTOR, 'input#name').send_keys(
                       'secrets'
```

**Pull request merged**